presto-client/src/utils.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Parses a JSON including bigger numbers into BigInts
+ * @param _ Key
+ * @param value Parsed value
+ * @param context Context with source text
+ * @returns Parsed object with BigInts where required
+ */
+export function parseWithBigInts(_: string, value: unknown, { source }: { source: string }) {
```

Comment:
Is it good to check the `arguments.length` here for compatibility? Then, there is no need for the `isJsonParseContextAvailable` function.

Reply:
The problem is `JSON.parse` always has 2 arguments: the actual text to be parsed, and the Reviver callback.
The difference between the old and newer specification is on how many arguments does JSON.parse pass to that callback (2 or 3). We can't easily check how many arguments will be there on a runtime callback, as far as I know.

Reply:
oh, just realized you mean checking for the arguments.length inside the reviver itself, that's a good idea! I'll try that 👍